content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,342 @@
+        break
+    time.sleep(1)
+```
+
+Output:
+
+```python
+Current status:  running
+...
+Current status:  finished
+```
+
+### Checking workflow logs
+
+If our workflow has failed or in case we want to debug or investigate how our
+workflow works, we may want to display the logs of our workflow. To do that
+using REANA client we could use [logs](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-logs)
+command. In our case the logs can be fetched with `get_workflow_logs` function
+where we only need to specify workflow name and access token. Optionally, we
+can add such parameters as a list of `steps` whose logs we are interested in
+or `page` and `size` parameters to paginate the logs output if it gets too
+verbose. For example, let's say that we want to quickly check the last 15 lines
+of logs of the second `fitdata` step of our workflow. We can do that as follows:
+
+```python
+import json
+from reana_client.api.client import get_workflow_logs
+
+workflow_logs = get_workflow_logs(my_workflow_name, my_reana_token, ['fitdata'])
+job_logs = json.loads(workflow_logs['logs'])['job_logs']
+fitdata_logs = job_logs[next(iter(job_logs))]['logs']
+
+for item in fitdata_logs.split('\n')[-15:]:
+    print(item)
+```
+
+Output:
+
+```python
+ERR MATRIX APPROXIMATE
+ PARAMETER  CORRELATION COEFFICIENTS
+       NO.  GLOBAL      1      2      3      4      5
+        1  0.00000   1.000  0.000  0.000  0.000  0.000
+        2  0.00000   0.000  1.000  0.000  0.000  0.000
+        3  0.00000   0.000  0.000  1.000  0.000  0.000
+        4  0.00000   0.000  0.000  0.000  1.000  0.000
+        5  0.00000   0.000  0.000  0.000  0.000  1.000
+ ERR MATRIX APPROXIMATE
+[#1] INFO:Minization -- RooMinimizer::optimizeConst: deactivating const optimization
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) directly selected PDF components: (bkg)
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) indirectly selected PDF components: ()
+Info in <TCanvas::Print>: png file results/plot.png has been created
+```
+
+### Listing workspace files
+
+Once the workflow is finished, we usually want to list all the output files which our workflow
+produced. With REANA client it can be achieved by using
+[ls](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-ls)
+command. Similarly, it can be done in the following manner:
+
+```python
+from reana_client.api.client import list_files
+
+list_files(my_workflow_name, my_reana_token)
+```
+
+Output:
+
+```python
+[
+  {
+    'last-modified': '2021-02-24T09:57:01',
+    'name': 'code/gendata.C',
+    'size': 1937,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:01',
+    'name': 'code/fitdata.C',
+    'size': 1648,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:52',
+    'name': 'results/plot.png',
+    'size': 15450,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:42',
+    'name': 'results/data.root',
+    'size': 154458,
+  },
+]
+```
+
+### Downloading outputs
```

Comment:
We are now interested in downloading workflow outputs, or any temporary files from the workflow's workspace, for closer inspection. In command-line interface, we would use the `download` CLI command. In Python, the corresponding function is `download_file()` function. Let us download a binary blob of the final plot which our workflow produced:

...